secure_message/repository/retriever.py

```diff
@@ -242,6 +243,20 @@ def _retrieve_thread_for_internal_user(thread_id):
 
         return result
 
+    @staticmethod
+    def retrieve_conversation_metadata(thread_id):
+        result = Conversation.query.filter(Conversation.id == thread_id)
+        try:
+            return result.one()
+        except NoResultFound:
+            logger.info("No conversation found", thread_id=thread_id)
+            raise NotFound(description=f"Conversation with thread_id '{thread_id}' does not exist")
+        except MultipleResultsFound:
```

Comment:
I don't see how this would _ever_ happen if we're querying on the primary key. If we can't write a test to test this, we shouldn't have it in the code.

Reply:
It wouldn't happen.  I only put in there for completeness as .one() can throw two different exceptions.  If it's valid to not handle an exception that can't be thrown then that's fine by me as it's less code